### ISSUE TITLE
Omg do I hate the apple emojis!

### ISSUE THREAD

**Issue by JoeJJJoe** [WhisperSystems/TextSecure#3675]
I'm using a Google Android device and not an iPhone. I am used to the Google emojis and not the Apple emojis. At least make it an option to switch back please. This shit is so annoying.

**Comment by wp9015362** [comment]
I disagree.

As already mentioned in https://github.com/WhisperSystems/TextSecure/issues/3662, IMHO the Google "Noto Color Emoji" Emoji set looks like shi*.

Let's face it, most people (or at least a lot), do not like the Google "Noto Color Emoji" Emoji set.

If you don't believe me, then maybe you should read the comments in the following two issues on the AOSP issue tracker:

https://code.google.com/p/android/issues/detail?id=69755
https://code.google.com/p/android/issues/detail?id=73248

And maybe you should also read the comments in the following XDA-Developers forum thread:

http://forum.xda-developers.com/showthread.php?t=2722892

The Apple Emoji set looks a heck of a lot better and is much more popular than the Google "Noto Color Emoji" Emoji set.

The Apple Emoji set is being used in Telegram, ICQ, WhatsApp and so on... (just to name a few examples)

And even the Twitter "Twemoji" Emoji set and the Emoji One Emoji set clearly are modelled after the Apple Emoji set.

So, it's quite popular and therefore pretty much is the natural choice.

Also, Signal-iOS (the iOS counterpart to TextSecure) of course uses the Apple Emoji set, since all iOS devices are using it natively.

So, now that TextSecure for Android also uses the Apple Emoji set, the Emoji experience is much more consistent across both platforms, which lessens the risk of Emojis getting lost in translation as described in the following article for example:

http://www.huffingtonpost.com/2014/06/27/emoji-meaning_n_5530638.html

Regards

**Comment by JoeJJJoe** [comment]
None of this makes any sense:

You think the Google emojis look like sh*t? Yeah well, that's like totally your opinion man! I think they look awesome. But it doesn't even matter what either of us think, because on a Google Android device the Google emojis are the standard and should be used, and not the Apple emojis or any other emoji set. At least not without giving the option to switch back.

And oh noes, some stupid woman is too stupid to realize that emojis can look different on different devices? Oh the humanity! Fine, if we need to make sure some random woman isn't being confused by how emojis work, then make one set the standard on all platforms and give users the choice to switch back to their devices default! 

What a pointless, unnecessary and annoying "improvement".

**Comment by wp9015362** [comment]
> @JoeJJJoe wrote:
> 
> on a Google Android device the Google emojis are the standard

That is just plain wrong.

In fact most Android devices do not use the Google "Noto Color Emoji" Emoji set.

HTC Android devices are using their own HTC Emoji set, Samsung Android devices are using their own Samsung Emoji set, LG Android devices are using their own LG Emoji set, Google Android devices (Nexus) are using their own Google Emoji set and so on... (just to name a few)

So, in reality, the Google "Noto Color Emoji" Emoji set is not the standard Emoji set on Android devices.

**Comment by JoeJJJoe** [comment]
And neither are the Apple emojis the standard. But I guess we need to dumb it down so the iphone users don't get confused. Whatever. This makes me seriously reconsider going back to Google Hangouts, even after I bothered everyone else to swtich to Textsecure. Who cares if Google reads my conversations, at least I don't have to deal with this crap.

**Comment by johanw666** [comment]
Hey, noone prevents you from getting the TextSecure code, put the Google emojis back in and compile it yourself. I do that myself to make some changes I prefer, like not interfering with SMSSecure and puting the encrypted backup back in.

You just need to revert https://github.com/WhisperSystems/TextSecure/commit/9a18be0a6194d4dd8ef420e9b567b1d7ce69c613 and https://github.com/WhisperSystems/TextSecure/commit/282f139ee9f9483a49ac78f448d1e2d3bbb5e374 , that's all.

**Comment by JoeJJJoe** [comment]
Just because I can undo a stupid change doesn't mean it's suddenly not stupid anymore. It's really fucking simple: On a Google Android device I expect to see the Google emojis, and not the Apple emojies. It's really not that hard to see this point, is it?

Give the user a choice if you feel so strongly about "unifying the experience". Not everyone wants to see the same Apple crap everywhere. Thank you.

**Comment by johanw666** [comment]
Tell that to WhatsApp.

**Comment by JoeJJJoe** [comment]
No, I'm telling it to WhisperSystems.

**Comment by wp9015362** [comment]
> @JoeJJJoe wrote:
> 
> And neither are the Apple emojis the standard.

The Apple Emoji set is the de facto standard.

WhatsApp uses it (on all major platforms)
Telegram uses it (on all major platforms)
Threema uses it (on all major platforms)
ICQ uses it (on all major platforms)

And even GitHub uses it.

Just to name a few examples... there are probably more...

Twitter also uses it's own open source "Twemoji" set which clearly is modelled after the Apple Emoji set.

If you don't believe that, then just have a look at the following comparision:

http://mroth.github.io/emojicompare/

Regards

**Comment by johanw666** [comment]
They prefer the Apple emojis. The advantage of open source is that if you don't agree you can change it and built your own version. The change is quite easy. It's their program, and I don't agree with all decicions they make either but I don't whine about it and sove the issues that are important to me.

**Comment by JoeJJJoe** [comment]
Google Hangouts doesn't use it. And that's the "de facto standard" for messaging on any Google Android device. Anyway, this discussion is as pointless as this change. Whatever. I said all I have to say.

**Comment by johanw666** [comment]
Noone uses Hangouts. I even freezed it so it doesn't consume background resources.

**Comment by JoeJJJoe** [comment]
Who cares what you did.

**Comment by ncruces** [comment]
Consistency isn't a valid argument for an app that does SMS. There's no way of being consistent for SMS, nor is there a point on even trying to be consistent when what you're implementing is a Unicode that explicitly allows such variations.

Even if you don't like Google emoji, there are free and open-source alternatives to Apple, that work better with the design of TextSecure. So, what exactly is the copyright on images added in eef0bcae5075d0419838b74fc8fc82bba7d98698?

**Comment by wp9015362** [comment]
> @ncruces wrote:
> 
> Even if you don't like Google emoji, there are free and open-source alternatives to Apple, that work better with the design of TextSecure.

Well, actually I have asked the TextSecure developers several times to use the "Emoji One" Emoji set:

http://emojione.com/
https://github.com/Ranks/emojione

But they simply changed to the Apple Emoji set instead, see for yourself in the following issue for example:

https://github.com/WhisperSystems/TextSecure/issues/3445

Regards

**Comment by Dyras** [comment]
Personally I would also prefer Emoji One. I think they look nicer, they have every flag in existence and they are free, which means no arguing five years from now when Apple flips over apps using their emoji on Android devices.

I'm still happier with Apple's emojis than I ever were with Google's potato emojis.

**Comment by wp9015362** [comment]
> @Dyras wrote:
> 
> Personally I would also prefer Emoji One. I think they look nicer, they have every flag in existence and they are free

And the Emoji One developers even already said that they will release a Unicode 7.0 update soon (which, according to them, will even include the middle finger Emoji), see the following comments for example:

https://github.com/Ranks/emojione/issues/68#issuecomment-93861096
https://github.com/Ranks/emojione/issues/68#issuecomment-93880415
https://github.com/Ranks/emojione/issues/105#issuecomment-106951709

:wink:

Regards

**Comment by entonjackson** [comment]
Please make it configurable, which emojis to choose. I prefer Google ones a lot more. And it seems there is bigger crowd sharing this opinion.

**Comment by wp9015362** [comment]
Looks like there are some more users who would prefer the Emoji One Emoji set, see the following comments over there:

https://github.com/WhisperSystems/TextSecure/pull/3532

>  entonjackson@ wrote:
> 
> Please make it configurable, which emojis to choose.

Well, the TextSecure developers said:

> @mcginty wrote in https://github.com/WhisperSystems/TextSecure/pull/3532#issuecomment-119007075:
> 
> we decided emoji are something we want to be stable across platforms, so we're standardizing on iOS emoji

So, letting the users choose would make the Emoji experience more unstable.

It can also lead to the issue where Emojis get lost in translation, as described in the following article:

http://www.huffingtonpost.com/2014/06/27/emoji-meaning_n_5530638.html

I am against an option to let the user choose.

I want to be sure that when I send someone an Emoji, that he/she sees the same Emoji that I see.

So, please do not add an option which let's the user choose different Emoji sets.

It should always be the same Emoji set across all platforms IMHO.

Regards

**Comment by JoeJJJoe** [comment]
At the very least add on option to use the device's system set, whichever that may be, because that's the one that is supposed to be used on that device in order for the user of that device to have a consistent experience on his device.

**Comment by wp9015362** [comment]
No, as already mentioned in the previous post, please don't do what @JoeJJJoe suggests.

> @JoeJJJoe wrote:
> 
> in order for the user of that device to have a consistent experience on his device.

That is not really an issue. People are already used to it when they use WhatsApp, Telegram, Threema and so on...

Instead of bothering TextSecure developers to bring configurable Emojis, you might rather want to participate in the following issue on the AOSP issue tracker:

https://code.google.com/p/android/issues/detail?id=69755

Because Android does not allow to change the system Emoji font without root. And even then it can be buggy.

So, if you want to have configurable system Emojis, then you need to ask the AOSP developers, not the TextSecure developers.

I also would like it if Android would have an official option to change the native system Emoji font. But unfortunately it doesn't.

Until that hasn't changed, the TextSecure Emoji approach is preferrable IMHO (overwriting native system Emojis with TextSecure built-on Emoji set).

Regards

**Comment by JoeJJJoe** [comment]
People like you are the worst.

**Comment by entonjackson** [comment]
Let's see what people think:
http://doodle.com/42wdk33m39uatpz8#table

**Comment by wp9015362** [comment]
> @entonjackson wrote:
> 
> Let's see what people think:
> http://doodle.com/42wdk33m39uatpz8#table

Just saying:

Your survey does not include an option for the Apple Emoji set (which is the one that TextSecure is currently using in version 2.22.2)...

Regards

**Comment by entonjackson** [comment]
I thought, that's the Emoji One thing??

**Comment by wp9015362** [comment]
> @entonjackson wrote:
> 
> I thought, that's the Emoji One thing??

No.

Before version 2.22.x, TextSecure was using the the Google Emoji set.

Now, starting with version 2.22.x, TextSecure is using the Apple Emoji set.

The Emoji One set is yet another Emoji set.

You can see a comparison of those Emoji sets (including most other Emoji sets) on the following page:

http://emoji.codes/family

As you can see on that page, the Emoji One Emoji set looks similar to the Apple Emoji set (just like the Twitter Twemoji Emoji set looks similar).

This is actually a design goal of Emoji One, see the following comment from a Emoji One developer:

https://github.com/Ranks/emojione/issues/68#issuecomment-93861096

Regards

**Comment by johanw666** [comment]
If the design goal of the emoji one set is to look like the Apple emoji, then why not go for the real thing and pick the Apple emojis? The only thing I can imagine preventing this are copyright issues but I didn't get any reply on my questions about that.

**Comment by wp9015362** [comment]
WOW!

I just noticed that Emoji One actually has release the Unicode 7.0 update today, see:

http://emojione.com/
https://github.com/Ranks/emojione/commit/aa55753abfc41b64141c0f033672dc525df4671d
https://github.com/Ranks/emojione/commit/bc777cac0e495a18ac4a9b0e794be5b1a111d1ca
https://github.com/Ranks/emojione/commit/4e995d819d47908f8777237a08df4702e5ac7d01
https://github.com/Ranks/emojione/commit/45148d1a82ad377cdcdd4598963db7fe1e939512

So, Emoji One is now Unicode 7.0 compatible (including middle finger) :laughing:.

:wink:

Regards

**Comment by JoeJJJoe** [comment]
If the only choice is between this Emoji One set and the Apple set, then obviously thet Apple set is the better choice: Both completely break the system's default, but at least other apps also use the Apple set when breaking the system's default. 

The only good solution here is to make one set the default on all platform and give the user a checkbox choice to enable the system's default emojis.

**Comment by ncruces** [comment]
Or use something like [twemoji](http://twitter.github.io/twemoji/), which is opensource, used by Twitter, more consistent with Apple, and has a design that better matches the TextSecure app.

**Comment by JoeJJJoe** [comment]
Yeah, it really doesn't matter that much as long as there is an option to fall back to the device's system default.

**Comment by wp9015362** [comment]
> @ncruces wrote:
> 
> Or use something like twemoji, which is opensource, used by Twitter, more consistent with Apple, and has a design that better matches the TextSecure app.

Twemoji was not really designed by Twitter AFAICT.

Twitter hired "The Iconfactory" to design Twemoji for them, see for yourself:

http://design.iconfactory.com/twitter-emoji/

Then Twitter opensourced Twemoji.

But now Twemoji seems to be unmaintained. AFAICT Twitter does not seem to care about Twemoji, or at least they do not reply to any issues on GitHub, see for yourself:

https://github.com/twitter/twemoji/issues

Twemoji also is not Unicode 7.0 compatible.

Emoji One on the other handy is also open source and is actively maintained. As already mentioned, Emoji One released their big Unicode 7.0 update today.

So, AFAICT, Emoji One is a much better choice than Twemoji.

Regards

**Comment by moxie0** [comment]
this isn't a discussion forum
